### PR TITLE
Expose binding secret without requiring a workload

### PR DESCRIPTION
**Pull request by sadlerap**
Signed-off-by: Andy Sadler <ansadler@redhat.com>

# Changes

/kind enhancement

This exposes the relevant binding secret without requiring a workload to be present.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#docs) 
  included if any changes are user facing
- [x] [Tests](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#tests)
  included if any functionality added or changed. For bugfixes please include tests that can catch regressions
- [x] All acceptance test scenarios included in the PR which verifies a bugfix or a requested feature reported by a non-member are tagged with `@external-feedback` tag.
- [x] Follows the [commit message standard](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#commits)

### CODE REVIEW

File: pkg/reconcile/pipeline/context/spec_binding_impl_test.go [hunk 1]
```diff
@@ -779,78 +779,43 @@ var _ = Describe("Spec API Context", func() {
...
+			u, _ := converter.ToUnstructured(&sb)
+			s := runtime.NewScheme()
+			Expect(specapi.AddToScheme(s)).NotTo(HaveOccurred())
+			Expect(corev1.AddToScheme(s)).NotTo(HaveOccurred())
+			client = fake.NewSimpleDynamicClient(s, u)
 
-			Expect(u.Object["Spec"]).To(Equal(specData))
+			authClient := &fakeauth.FakeAuthorizationV1{}
 
+			ctx, _ = Provider(client, authClient.SubjectAccessReviews(), typeLookup).Get(sb)
```
Comment (baijum) [inline pkg/reconcile/pipeline/context/spec_binding_impl_test.go:818]: I don't any assertion after this line (No `Expect(...)` calls).

File: pkg/reconcile/pipeline/context/impl_test.go [hunk 2]
```diff
@@ -878,22 +857,40 @@ var _ = Describe("Context", func() {
...
+			}
+			sb.SetGroupVersionKind(bindingapi.GroupVersionKind)
+			u, _ := converter.ToUnstructured(&sb)
+			s := runtime.NewScheme()
+			Expect(bindingapi.AddToScheme(s)).NotTo(HaveOccurred())
+			Expect(corev1.AddToScheme(s)).NotTo(HaveOccurred())
+			client = fake.NewSimpleDynamicClient(s, u)
+			authClient := &fakeauth.FakeAuthorizationV1{}
+
+			ctx, _ = Provider(client, authClient.SubjectAccessReviews(), typeLookup).Get(sb)
```
Comment (baijum) [inline pkg/reconcile/pipeline/context/impl_test.go:891]: I don't any assertion after this line (No `Expect(...)` calls).

**Review by baijum — APPROVED** [review]
/lgtm

File: test/acceptance/features/bindAppToServiceAnnotations.feature [hunk 3]
```diff
@@ -726,3 +726,37 @@ Feature: Bind an application to a service using annotations
...
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: $scenario_id-backend
+                annotations:
+                    service.binding/host_internal_db: path={.spec.host_internal_db}
+            spec:
+                host_internal_db: internal.db.stable.example.com
+            """
```
Comment (pmacik) [inline test/acceptance/features/bindAppToServiceAnnotations.feature:742]: If the binding secret is to be generated without the workload, then the `Backend` CR (the workload) should not be created, should it?

If that is the case, than these two steps should be removed from the scenario

**Pull request merged**